client/__tests__/Comparison.test.js

```diff
@@ -0,0 +1,173 @@
+        'url': 'https://images.unsplash.com/photo-1534550017194-5df79ed78967?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1000&q=80'
+      },
+      {
+        'thumbnail_url': 'https://images.unsplash.com/photo-1525896650794-60ad4ec40d0e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80',
+        'url': 'https://images.unsplash.com/photo-1525896650794-60ad4ec40d0e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80'
+      },
+      {
+        'thumbnail_url': 'https://images.unsplash.com/photo-1560857792-215f9e3534ed?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80',
+        'url': 'https://images.unsplash.com/photo-1560857792-215f9e3534ed?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80'
+      }
+    ],
+    'skus': {
+      '714558': {
+        'quantity': 14,
+        'size': '7'
+      },
+      '714559': {
+        'quantity': 25,
+        'size': '7.5'
+      },
+      '714560': {
+        'quantity': 9,
+        'size': '8'
+      },
+      '714561': {
+        'quantity': 2,
+        'size': '8.5'
+      },
+      '714562': {
+        'quantity': 18,
+        'size': '9'
+      },
+      '714563': {
+        'quantity': 12,
+        'size': '9.5'
+      },
+      '714564': {
+        'quantity': 10,
+        'size': '10'
+      },
+      '714565': {
+        'quantity': 18,
+        'size': '10.5'
+      },
+      '714566': {
+        'quantity': 11,
+        'size': '11'
+      },
+      '714567': {
+        'quantity': 35,
+        'size': '11.5'
+      },
+      '714568': {
+        'quantity': 25,
+        'size': '12'
+      }
+    }
+  }
+};
+
+const relatedProductInformation = JSON.parse(JSON.stringify(currentProductInformation));
+relatedProductInformation.features.push({
+  feature: 'this feature is only on the related product',
+  value: true
+});
+
+beforeEach(() => {
+  render(<Comparison
+    currentProduct={currentProductInformation}
+    relatedProduct={relatedProductInformation}
+  />);
+});
+
+afterEach(() => {
+  cleanup();
+});
+
+test('Should render Comparison', () => {
+  const ComparisonComponent = screen.getByTestId('comparison');
+  expect(ComparisonComponent).toBeInTheDocument();
+});
+
+test('Should contain product titles', () => {
+  const currentTitle = screen.getByTestId('comparison--current-title');
+  const relatedTitle = screen.getByTestId('comparison--related-title');
+  expect(currentTitle).toBeInTheDocument();
+  expect(relatedTitle).toBeInTheDocument();
+});
+
+test('Should contain traits', () => {
+  // This should contain 5 traits, as there are four that are shared and one that is unique
+  const comparisonTraits = screen.getAllByTestId('comparison-characteristics__characteristic');
+  expect(comparisonTraits.length).toBe(5);
+});
+
+test('Should render a check mark if feature attribute is boolean', () => {
```

Comment:
is boolean? Is present, does this mean? 
This is as close as I get to a change request - this test description is a little opaque.

Reply:
Good point. They are naturally just a string, but they mention in the doc that if the attribute is true or false/yes or no, use a check mark. I have yet to experience one, so am assuimng this is how they will be presented, but I'll clarify the tests further for future readers.